astropy/wcs/wcsapi/fitswcs.py

```diff
@@ -121,6 +121,13 @@ def pixel_n_dim(self):
     def world_n_dim(self):
         return len(self.wcs.ctype)
 
+    @property
+    def array_shape(self):
```

Comment:
It's in array/numpy order - all these properties have docstrings in the base API that get inherited, that's why there is apparently no docstring here (but users would see one). See https://github.com/astropy/astropy/blob/master/astropy/wcs/wcsapi/low_level_api.py#L196

In these docs, I've also tried to clarify that anything with 'array' in the name is in array index order.

Reply:
Note that we could add ``pixel_shape`` if you wanted

Reply:
:+1: for adding `pixel_shape` because its order corresponds to the `NAXIS*` order. 

I think we should deprecate the setters for the private `_naxis1/2` attributes and change `calc_footprint` to use `pixel_shape`. I can do this in a separate PR if this sounds acceptable.

Reply:
I'm also 👍 to adding `pixel_shape`.  While it seems a bit picky it seems consistent with the other methods now.

Reply:
Added